feat(legacy-preset-chart-deckgl): Add ,.1f and ,.2f value formats to deckgl charts

kgabryje opened the pull request:
### SUMMARY
This PR adds 2 new formattings to Legend Format control in deckgl charts.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<img width="584" alt="image" src="https://user-images.githubusercontent.com/15073128/155715871-4acf6a01-86ce-4ad5-a0b7-d18a74200773.png">

### TESTING INSTRUCTIONS
Verify that ,.1f and ,.2f legend formattings are available and work correctly

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

https://app.shortcut.com/preset/story/37656/add-1f-and-2f-to-the-legend-format-option-for-deck-gl-charts

kgabryje committed:
Add ,.1f and ,.2f value formats to deckgl charts

michael-s-molina reviewed (approved):
LGTM

zhaoyongjie commented on superset-frontend/plugins/legacy-preset-chart-deckgl/src/utilities/controls.jsx:
```diff
@@ -37,6 +37,8 @@ export const D3_FORMAT_OPTIONS = [
   [',.1%', ',.1% (12345.432 => 1,234,543.2%)'],
   ['.3%', '.3% (12345.432 => 1234543.200%)'],
   ['.4r', '.4r (12345.432 => 12350)'],
+  [',.1f', ',.1f (12345.432 => 12,345.4)'],
```
﻿﻿﻿I remember that users can enter d3format manually. Why do we need to create a format in deckgl that is not consistent with all other charts?

<img width="1115" alt="image" src="https://user-images.githubusercontent.com/2016594/155721403-170e5cc7-b1a9-4c9f-9714-b9896291b13e.png">

zhaoyongjie reviewed (changes requested):
In order to provide more format for this chart,  we just add `freeForm: true` for format control in Deckgl.

kgabryje commented:
Thanks for pointing that out @zhaoyongjie. @yousoph @rusackas should we add those formatters in other charts as well? Should we quick and simple.

yousoph commented:
Thanks for pointing that out, @zhaoyongjie ! 
@kgabryje I think it makes sense to add the freeform formatter to deck.gl (and any other chart types it's missing from)

kgabryje committed:
Remove duplicated code

kgabryje commented:
@zhaoyongjie @yousoph Added `,.1f` and `,.2f` to `D3_FORMAT_OPTIONS` globally, added `freeForm` to deckgl, also removed duplicated code in deckgl plugins.

zhaoyongjie reviewed (approved):
Thanks for the improvement!
